src/main/java/jp/co/soramitsu/sora/common/ArrayTree.java

```diff
@@ -0,0 +1,96 @@
+package jp.co.soramitsu.sora.common;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Getter;
+
+public class ArrayTree<E> {
+
+  @Getter
+  private List<E> tree;
+
+  public ArrayTree(List<E> tree) {
```

Comment:
assert that tree size is `size + 1 == powerOf2()`, or we will meet inconsistencies across all methods

Reply:
just make sure all bits except highest are zero or absent